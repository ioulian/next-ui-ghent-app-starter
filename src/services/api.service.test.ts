/* eslint-disable i18next/no-literal-string */

import { getContentTypeHeaders, injectHeaders } from "@/services/api.service";

describe("api.service", () => {
  it("injectHeaders works correctly", () => {
    expect(injectHeaders({ Authorization: "1" })).toMatchSnapshot();
    expect(injectHeaders({ Authorization: "1" }, { headers: {} })).toMatchSnapshot();
    expect(
      injectHeaders({ Authorization: "1" }, { headers: { "Content-Type": "application/json" } }),
    ).toMatchSnapshot();
    expect(
      injectHeaders({ Authorization: "1" }, { headers: { Authorization: "none" } }),
    ).toMatchSnapshot();
  });

  it("getContentTypeHeaders works correctly", () => {
    expect(getContentTypeHeaders()).toMatchSnapshot();
  });
});
