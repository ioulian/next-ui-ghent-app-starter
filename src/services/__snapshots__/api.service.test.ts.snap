// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api.service injectHeaders works correctly 1`] = `
{
  "headers": {
    "Authorization": "1",
  },
}
`;

exports[`api.service injectHeaders works correctly 2`] = `
{
  "headers": {
    "Authorization": "1",
  },
}
`;

exports[`api.service injectHeaders works correctly 3`] = `
{
  "headers": {
    "Authorization": "1",
    "Content-Type": "application/json",
  },
}
`;

exports[`api.service injectHeaders works correctly 4`] = `
{
  "headers": {
    "Authorization": "none",
  },
}
`;
